cmd/server/pactasrv/initiative.go

```diff
@@ -0,0 +1,164 @@
+	err := s.updateInitiative(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.UpdateInitiativedefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.UpdateInitiative200JSONResponse(emptySuccess), nil
+}
+
+func (s *Server) updateInitiative(ctx context.Context, request api.UpdateInitiativeRequestObject) error {
+	// TODO(#12) Implement Authorization
+	id := pacta.InitiativeID(request.Id)
+	mutations := []db.UpdateInitiativeFn{}
+	b := request.Params.Body
+	if b.Affiliation != nil {
+		mutations = append(mutations, db.SetInitiativeAffiliation(*b.Affiliation))
+	}
+	if b.InternalDescription != nil {
+		mutations = append(mutations, db.SetInitiativeInternalDescription(*b.InternalDescription))
+	}
+	if b.IsAcceptingNewMembers != nil {
+		mutations = append(mutations, db.SetInitiativeIsAcceptingNewMembers(*b.IsAcceptingNewMembers))
+	}
+	if b.IsAcceptingNewPortfolios != nil {
+		mutations = append(mutations, db.SetInitiativeIsAcceptingNewPortfolios(*b.IsAcceptingNewPortfolios))
+	}
+	if b.Language != nil {
+		lang, err := pacta.ParseLanguage(string(*b.Language))
+		if err != nil {
+			return errorBadRequest("language", err.Error())
+		}
+		mutations = append(mutations, db.SetInitiativeLanguage(lang))
+	}
+	if b.Name != nil {
+		mutations = append(mutations, db.SetInitiativeName(*b.Name))
+	}
+	if b.PactaVersion != nil {
+		mutations = append(mutations, db.SetInitiativePACTAVersion(pacta.PACTAVersionID(*b.PactaVersion)))
+	}
+	if b.PublicDescription != nil {
+		mutations = append(mutations, db.SetInitiativePublicDescription(*b.PublicDescription))
+	}
+	if b.RequiresInvitationToJoin != nil {
+		mutations = append(mutations, db.SetInitiativeRequiresInvitationToJoin(*b.RequiresInvitationToJoin))
+	}
+	err := s.DB.UpdateInitiative(s.DB.NoTxn(ctx), id, mutations...)
+	if err != nil {
+		return errorInternal(err)
+	}
+	return nil
+}
+
+// Deletes an initiative by id
+// (DELETE /initiative/{id})
+func (s *Server) DeleteInitiative(ctx context.Context, request api.DeleteInitiativeRequestObject) (api.DeleteInitiativeResponseObject, error) {
+	err := s.deleteInitiative(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.DeleteInitiativedefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.DeleteInitiative200JSONResponse(emptySuccess), nil
+}
+
+func (s *Server) deleteInitiative(ctx context.Context, request api.DeleteInitiativeRequestObject) error {
+	// TODO(#12) Implement Authorization
+	err := s.DB.DeleteInitiative(s.DB.NoTxn(ctx), pacta.InitiativeID(request.Id))
+	if err != nil {
+		return errorInternal(err)
+	}
+	return nil
+}
+
+// Returns an initiative by ID
+// (GET /initiative/{id})
+func (s *Server) FindInitiativeById(ctx context.Context, request api.FindInitiativeByIdRequestObject) (api.FindInitiativeByIdResponseObject, error) {
+	result, err := s.findInitiativeById(ctx, request)
```

Comment:
I understand the exported/unexported structure is to capture all the errors in one place, but I think with the structure I'm proposing, we shouldn't need it. Errors would just be handled explicitly inline